### DISCUSSION TITLE
How to use a deployment renv profile

### DISCUSSION THREAD

**Discussion by klmr** [Appsilon/rhino#622] [Q&A]
Does Rhino support [‘renv’ profiles](https://rstudio.github.io/renv/articles/profiles.html)?

As far as I can tell, Rhino uses a single file, `dependencies.R`, to maintain dependencies via ‘renv’. Would there be a way to distinguish between dev and deployment dependencies?

For instance, I certainly don’t want to deploy ‘treesitter’, ‘treesitter.r’ or ‘shinytest2’ — at best it’s an unnecessary overhead. However, the way ‘renv’ is configured it seems that `dependencies.r` is effectively hard-coded…

‘renv’ also has a separate concept of [development dependencies](https://rstudio.github.io/renv/reference/dependencies.html#development-dependencies) but that requires using a `DESCRIPTION` file, and it’s also a bit weird, since it doesn’t snapshot these dependencies at all, which seems problematic.

**Comment by jakubnowicki** [comment]
Unfortunately, `rhino` does not have a mechanism for development dependencies. But this is one of the topics we definitely want to address in the future releases (we were thinking about this for a long time, e.g., #343 ).
